### PR TITLE
pd: check cluster id while connecting to leader

### PR DESCRIPTION
**Pull request by overvenus**
PTAL @BusyJay @siddontang @disksing

### CODE REVIEW

File: src/pd/async/util.rs [hunk 1]
```diff
@@ -311,13 +311,19 @@ pub fn try_connect_leader(previous: &GetMembersResponse)
...
         for ep in members[i].get_client_urls() {
             match connect(ep.as_str()) {
                 Ok((_, r)) => {
-                    resp = Some(r);
-                    break 'outer;
+                    if r.get_header().get_cluster_id() == cluster_id {
+                        resp = Some(r);
+                        break 'outer;
+                    } else {
+                        warn!("{} no longer belongs to cluster {}", ep, cluster_id);
```
Comment (BusyJay) [inline src/pd/async/util.rs]: New cluster id should be printed too.

Comment (siddontang) [inline src/pd/async/util.rs]: should panic here directly

File: src/pd/async/util.rs [hunk 2]
```diff
@@ -311,13 +311,23 @@ pub fn try_connect_leader(previous: &GetMembersResponse)
...
             match connect(ep.as_str()) {
                 Ok((_, r)) => {
-                    resp = Some(r);
-                    break 'outer;
+                    let new_cluster_id = r.get_header().get_cluster_id();
+                    if new_cluster_id == cluster_id {
+                        resp = Some(r);
+                        break 'outer;
+                    } else {
+                        warn!("{} no longer belongs to cluster {}, it is in {}",
```
Comment (siddontang) [inline src/pd/async/util.rs]: log error and exit directly.

**Comment by disksing** [comment]
LGTM.

**Comment by BusyJay** [comment]
LGTM

File: src/pd/async/util.rs (hunk 2 above)
Comment (siddontang) [inline src/pd/async/util.rs]: any update?

**Comment by siddontang** [comment]
PTAL @disksing

**Pull request merged**